cpp/examples/tpch/q5.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/ast/expressions.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+
+#include "utils.hpp"
+
+/*
+create view customer as select * from '~/tpch_sf1/customer/part-0.parquet';
+create view orders as select * from '~/tpch_sf1/orders/part-0.parquet';
+create view lineitem as select * from '~/tpch_sf1/lineitem/part-0.parquet';
+create view supplier as select * from '~/tpch_sf1/supplier/part-0.parquet';
+create view nation as select * from '~/tpch_sf1/nation/part-0.parquet';
+create view region as select * from '~/tpch_sf1/region/part-0.parquet';
+
+select
+    n_name,
+    sum(l_extendedprice * (1 - l_discount)) as revenue
+from
+    customer,
+    orders,
+    lineitem,
+    supplier,
+    nation,
+    region
+where
+        c_custkey = o_custkey
+  and l_orderkey = o_orderkey
+  and l_suppkey = s_suppkey
+  and c_nationkey = s_nationkey
+  and s_nationkey = n_nationkey
+  and n_regionkey = r_regionkey
+  and r_name = 'ASIA'
+  and o_orderdate >= date '1994-01-01'
+  and o_orderdate < date '1995-01-01'
+group by
+    n_name
+order by
+    revenue desc;
+*/
+
+std::unique_ptr<cudf::column> calc_revenue(std::unique_ptr<table_with_cols>& table) {
+    auto one = cudf::fixed_point_scalar<numeric::decimal64>(1, -2);
+    auto disc = table->column("l_discount");
+    auto one_minus_disc = cudf::binary_operation(one, disc, cudf::binary_operator::SUB, disc.type());
+    auto extended_price = table->column("l_extendedprice");
+
+    auto disc_price_type = cudf::data_type{cudf::type_id::DECIMAL64, -4};
+    auto disc_price = cudf::binary_operation(extended_price, one_minus_disc->view(), cudf::binary_operator::MUL, disc_price_type);
+    return disc_price;
+}
+
+int main(int argc, char const** argv) {
+    auto args = parse_args(argc, argv);
+
+    // Use a memory pool
+    auto resource = create_memory_resource(args.use_memory_pool);
+    rmm::mr::set_current_device_resource(resource.get());
+
+    Timer timer;
+
+    // 1. Read out the tables from parquet files
+    //    while pushing down column projections and filter predicates
```

Comment:
I would suggest removing the numbers like `// 1.` and just list the steps with no numbers. That makes it easier to refactor in the future.